src/web-extension/options/components/form.test.jsx

```diff
@@ -68,84 +79,85 @@ describe('form', () => {
 
-    const { instance } = render({ store });
-    jest.spyOn(instance, 'handleLoaded');
+    let loaded;
 
-    instance.componentDidMount();
+    store.get.mockReturnValue(new Promise((resolve) => {
+      // Create a new async function that resolves the promise returned from
+      // calls to `store.get()` and returns a promise itself which we can wait
+      // on to ensure the store results have been handled.
+      loaded = async function load(data) { resolve(data); };
+    }));
 
-    expect(store.get).toHaveBeenCalledWith(null, instance.handleLoaded);
-    expect(instance.handleLoaded).toHaveBeenCalledWith(data);
-  });
+    const wrapper = render({ store });
 
-  it('updates the form inputs once templates are loaded', () => {
-    const { wrapper, instance } = render({});
-    const data = { templates: { branch: 'x', commit: 'y', command: 'z' } };
+    expect(store.get).toHaveBeenCalledWith(null);
+    expect(inputs(wrapper).every({ disabled: true })).toBe(true);
 
-    instance.handleLoaded(data);
-    wrapper.update();
+    await loaded({ templates: { branch: 'a', commit: 'b', command: 'c' } });
 
-    expect(value(wrapper, 'branch')).toBe('x');
-    expect(value(wrapper, 'commit')).toBe('y');
-    expect(value(wrapper, 'command')).toBe('z');
+    expect(value(wrapper, 'branch')).toBe('a');
+    expect(value(wrapper, 'commit')).toBe('b');
+    expect(value(wrapper, 'command')).toBe('c');
 
     expect(inputs(wrapper).every({ disabled: false })).toBe(true);
   });
 
   it('updates the form inputs on changes', () => {
-    const { wrapper, instance } = render({});
+    const wrapper = render({});
 
-    change(wrapper, instance, 'branch', 'branch++');
+    change(wrapper, 'branch', 'branch++');
     expect(value(wrapper, 'branch')).toBe('branch++');
 
-    change(wrapper, instance, 'commit', 'commit++');
+    change(wrapper, 'commit', 'commit++');
     expect(value(wrapper, 'commit')).toBe('commit++');
 
-    change(wrapper, instance, 'command', 'command++');
+    change(wrapper, 'command', 'command++');
     expect(value(wrapper, 'command')).toBe('command++');
   });
 
-  it('stores templates on submit', () => {
+  it('stores templates on submit and disables form elements while saving', async () => {
     const store = { get: jest.fn(), set: jest.fn() };
-    store.set.mockImplementation((_, fn) => fn());
 
-    const { wrapper, instance } = render({ store });
-    jest.spyOn(instance, 'handleSaved').mockReturnValue();
+    let saved;
+
+    const unchanged = { branch: 'branch', commit: 'commit', command: 'command' };
+    store.get.mockResolvedValue({ templates: unchanged });
+    store.set.mockReturnValue(new Promise((resolve) => {
+      // Create a new async function that resolves the promise returned from
+      // calls to `store.set()` and returns a promise itself which we can wait
+      // on to ensure data saving has been handled.
+      saved = async function save() { resolve(); };
+    }));
```

Comment:
Not sure if this is easily understood or whether/how to improve it.

Basically I liked being able to say `await saved()` further down below to trigger "saving" and wait for all dependent promises (i.e. those re-enabling the form inputs) to finish processing.

@lwassermann, in case you get a chance to take another look, I'd be happy about your feedback 🙂